File: src/main/java/org/gjt/jclasslib/structures/attributes/targettype/EmptyTargetInfo.java
/*
 This library is free software; you can redistribute it and/or
 modify it under the terms of the GNU General Public
 License as published by the Free Software Foundation; either
 version 2 of the license, or (at your option) any later version.
 */

package org.gjt.jclasslib.structures.attributes.targettype;

/**
 * Target info for a <tt>TypeAnnotation</tt> structure with empty content.
 */
public class EmptyTargetInfo extends TargetInfo {

    @Override
    public int getLength() {
        return 0;
    }

    @Override
    public String getVerbose() {
        return "<none>";
    }
}

File: modules/data/src/commonMain/kotlin/structures/attributes/targettype/EmptyTargetInfo.kt
/*
 This library is free software; you can redistribute it and/or
 modify it under the terms of the GNU General Public
 License as published by the Free Software Foundation; either
 version 2 of the license or (at your option) any later version.
 */

package org.gjt.jclasslib.structures.attributes.targettype

import org.gjt.jclasslib.io.DataInput
import org.gjt.jclasslib.io.DataOutput

/**
 * Target info for a TypeAnnotation structure with empty content.
 */
class EmptyTargetInfo : TargetInfo() {

    override val length: Int
        get() = 0

    override val verbose: String
        get() = "<none>"

    override fun readData(input: DataInput) {
    }

    override fun writeData(output: DataOutput) {
    }

    override val debugInfo: String
        get() = ""
}
